src/main/osd/osd.c

```diff
@@ -483,13 +483,19 @@ static void osdShowStats(uint16_t endBatteryVoltage)
     }
 
     if (osdStatGetState(OSD_STAT_TIMER_1)) {
-        osdFormatTimer(buff, false, (OSD_TIMER_SRC(osdConfig()->timers[OSD_TIMER_1]) == OSD_TIMER_SRC_ON ? false : true), OSD_TIMER_1);
-        osdDisplayStatisticLabel(top++, osdTimerSourceNames[OSD_TIMER_SRC(osdConfig()->timers[OSD_TIMER_1])], buff);
+        int src = OSD_TIMER_SRC(osdConfig()->timers[OSD_TIMER_1]);
+        if (src >= OSD_TIMER_SRC_COUNT)
+            src = 0;
+        osdFormatTimer(buff, false, (src == OSD_TIMER_SRC_ON ? false : true), OSD_TIMER_1);
+        osdDisplayStatisticLabel(top++, osdTimerSourceNames[src], buff);
     }
 
     if (osdStatGetState(OSD_STAT_TIMER_2)) {
-        osdFormatTimer(buff, false, (OSD_TIMER_SRC(osdConfig()->timers[OSD_TIMER_2]) == OSD_TIMER_SRC_ON ? false : true), OSD_TIMER_2);
-        osdDisplayStatisticLabel(top++, osdTimerSourceNames[OSD_TIMER_SRC(osdConfig()->timers[OSD_TIMER_2])], buff);
+        int src = OSD_TIMER_SRC(osdConfig()->timers[OSD_TIMER_2]);
+        if (src >= OSD_TIMER_SRC_COUNT)
+            src = 0;
+        osdFormatTimer(buff, false, (src == OSD_TIMER_SRC_ON ? false : true), OSD_TIMER_2);
+        osdDisplayStatisticLabel(top++, osdTimerSourceNames[src], buff);
```

Comment:
There is a risk of invalid values if somebody sets the parameter incorrectly through the CLI which can't validate the bit packed values. However the proper place to validate and correct this would be in `validateAndFixConfig()` in `fc/config.c` rather than doing it continuously every loop at runtime.

Reply:
Agree, that validating in one place would be better.
Btw, the code here is only executed when displaying post-flight statistics, not during the flight.

Reply:
Adding the validation logic to `validateAndFixConfig()` will solve both cases though (CMS and OSD stats) and the runtime logic is then unnecessary.

Also the Configurator changes should respect the API version and not allow invalid timer sources for the connected firmware version. So the new timer source is only allowed when API version >= 1.42